client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala

```diff
@@ -149,6 +167,43 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
       0,
       heartbeatIntervalMs,
       TimeUnit.MILLISECONDS)
+
+    changePartitionSchedulerThread.scheduleAtFixedRate(
+      new Runnable {
+        override def run(): Unit = {
+          try {
+            while (true) {
+              changePartitionRequests.asScala.foreach { case (shuffleId, requests) =>
+                requests.synchronized {
+                  changePartitionExecutors.submit {
+                    new Runnable {
+                      override def run(): Unit = {
+                        // For each partition only need handle one request
+                        val distinctPartitions = requests.asScala.map { case (_, request) =>
+                          request.asScala.toArray.head
+                        }.toArray
+                        batchHandleChangePartitions(
+                          distinctPartitions.head.applicationId,
+                          shuffleId,
+                          distinctPartitions)
+                      }
+                    }
+                  }
+                  // After put request into executor, clean current requests.
+                  requests.clear()
```

Comment:
Should not clear() here, batchHandleChangePartitions will remove

Reply:
> Should not clear() here, batchHandleChangePartitions will remove

Oh, yea,  first I use an external map to store change partition request then merge it in to origin map, remain this. Good catch.